From 489b229104c76651ff36fc5639384cf9dc6b8d7d Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Fri, 23 Dec 2016 03:19:18 +0500
Subject: [PATCH] CrossCompile.cmake: use target BuildVariables include for
 host

This is primarily OE specific where we'd like to report
the target build variables when checked through host
llvm-config because that is used for configuring
projects depending on LLVM.

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 cmake/modules/CrossCompile.cmake | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/cmake/modules/CrossCompile.cmake b/cmake/modules/CrossCompile.cmake
index 9c598a6..173aefd 100644
--- a/cmake/modules/CrossCompile.cmake
+++ b/cmake/modules/CrossCompile.cmake
@@ -22,7 +22,8 @@ function(llvm_create_cross_target_internal target_name toolchain buildtype)
         -DLLVM_TARGET_IS_CROSSCOMPILE_HOST=TRUE
     WORKING_DIRECTORY ${LLVM_${target_name}_BUILD}
     DEPENDS ${LLVM_${target_name}_BUILD}
-    COMMENT "Configuring ${target_name} LLVM...")
+    COMMENT "Configuring ${target_name} LLVM..."
+    COMMAND "${CMAKE_COMMAND}" "-E" "copy" "${CMAKE_SOURCE_DIR}/../build/tools/llvm-config/BuildVariables.inc" "${CMAKE_SOURCE_DIR}/../build/NATIVE/tools/llvm-config/BuildVariables.inc")
 
   add_custom_target(CONFIGURE_LLVM_${target_name}
                     DEPENDS ${LLVM_${target_name}_BUILD}/CMakeCache.txt)
-- 
1.9.1

